src/components/response_error.tsx

```diff
@@ -8,63 +8,30 @@ import Tile from './tile';
- */
-export const ERROR_CODE_TO_I18NC_TEMPLATE = {
-
-    // generic errors
-    "genericError": "There was a problem on the server, please try again. (Code: %{code})",
-    "apiResponseError": "There was a problem on the server, please try again.",
-
-    // generic field errors
-    "fieldError": "%{field_name} was invalid: %{title}.",
-    "lessThanMinimum": "%{field_name} does not have enough characters.",
-
-    // specific field errors
-    "emailInvalid": "Email address was not valid. Please use only lower-case characters and a full domain name.",
-    "fullNameInvalid": "Full name was not valid. Please use only latin characters, \
-        numbers and simple separators (single space, underline or hyphen).",
-    "usernameInvalid": "Username was not valid. Please use only latin characters, \
-        numbers and simple separators (single space, underline or hyphen).",
-    "passwordInvalid": "Password was not valid. Please use at least eight characters, \
-        at least one number and both lower and uppercase letters and special characters.",
-    "badPasswordRepeat": "Repeated password did not match original password.",
-    "dataUriImageInvalid": "Profile image was not valid. Please use a JPEG or PNG image."
-
-}
-
-/**
- * Map of all form field names that should be translated.
- */
-export const FIELD_NAME_TO_I18NC = {
-
-    // user data
-    "username": I18n.translate("form field username", "Username"),
-    "email": I18n.translate("form field email", "Email Address"),
-    "name": I18n.translate("form field name", "Full Name"),
-    "password": I18n.translate("form field password", "Password"),
-    "password_repeat": I18n.translate("form field password_repeat", "Repeat Password"),
-
-}
+export type ErrorMessageTemplates = {[key: string]: string};
+export type FieldTranslations = {[key: string]: string};
 
 /**
  * Gets a translated message for the given error.
  */
-export const getTranslatedError = (error: ResponseErrorObject): string => {
+export const getTranslatedError = (
+    error: ResponseErrorObject,
+    errorMessageTemplates: ErrorMessageTemplates,
+    fieldTranslations: FieldTranslations
+): string => {
     let params = { ...error }
     if (params.meta) {
         Object.keys(error.meta).forEach(key => {
             let value = error.meta[key]
-            if (key === "field_name" && FIELD_NAME_TO_I18NC[value]) {
-                value = FIELD_NAME_TO_I18NC[value]
+            if (key === "field_name" && fieldTranslations[value]) {
+                value = fieldTranslations[value]
             }
             params[key] = value
         });
     }
-    return I18n.format("error code", ERROR_CODE_TO_I18NC_TEMPLATE[error.code]
-        || ERROR_CODE_TO_I18NC_TEMPLATE["genericError"], params)
+    const errorMessage = errorMessageTemplates[error.code] || I18n.translate('error message default',
+        "There was a problem on the server, please try again. (Code: %{code})")
```

Comment:
Can we maybe include the title here? Something like `"There was an issue on the server: {title} (code {code})"`

Reply:
Done.